ddtrace/opentelemetry/tracer_provider.go

```diff
@@ -3,6 +3,21 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2023 Datadog, Inc.
 
+// Package opentelemetry provides a wrapper on top of the Datadog tracer that can be used with OpenTelemetry.
+// It also provides a wrapper around TracerProvider to propagate a list of tracer.StartOption
+// that are specific to Datadog's APM product. To use it, simply call "NewTracerProvider".
+//
+//	provider := opentelemetry.NewTracerProvider(tracer.WithService("opentelemetry_service"))
+//
+// When using Datadog, the OpenTelemetry span name is what is called operation name in Datadog's terms.
+// To start the span with Datadog OpenTelemetry package, onw would do:
+//
+//	otel.SetTracerProvider(opentelemetry.NewTracerProvider())
+//	tracer := otel.Tracer("")
+//	ctx, sp := tracer.Start(context.Background(), "span_name")
+//
+// Note that there are currently some small incompatibilities between the OpenTelemetry spec and the Datadog
+// APM product, which we are in the process of addressing on the long term.
```

Comment:
I would call out which incompatibilities are there, so we've set expectations appropriately. i.e. if something is not working that we haven't listed isn't supported, then they should file an escalation. But otherwise, they don't need to if it's a known issue.

Could instead say something more along the lines of this:

"Not every feature provided by OpenTelemtry is supported with this wrapper today. This package seeks to implement a minimal set of functions within the OpenTelemetry Tracing API (https://opentelemetry.io/docs/reference/specification/trace/api) to allow users to send traces to Datadog using existing OpenTelemtry code with minimal changes to the application. Span events (https://opentelemetry.io/docs/concepts/signals/traces/#span-events) are not supported at this time."